reference/reference_test.go

```diff
@@ -96,6 +96,23 @@ func TestReferenceParse(t *testing.T) {
 			err:   digest.ErrDigestUnsupported,
 		},
 		{
+			input: "Uppercase:tag",
+			err:   ErrNameContainsUppercase,
+		},
+		{
+			input: "Uppercase/lowercase:tag",
+			err:   ErrNameContainsUppercase,
```

Comment:
@thaJeztah: See https://github.com/docker/docker/pull/20175. I remember there was a lot of discussion about how to fix this properly, and there was no solution that we were completely happy with for 1.10.1. As you suggest, this reference package doesn't distinguish between hostnames and image names when it parses, but doing so is necessary to match the behavior of Docker < 1.10.

There is work underway at #1778 to correct this.